src/q&a/components/QuestionList.jsx

```diff
@@ -8,52 +8,77 @@ import AddQuestion from './AddQuestionModal.jsx';
 function QuestionList({ props, handleQuestionSubmit }) {
   const [list, setList] = useState([]);
   const [matches, setMatches] = useState([]);
+  const [index, setIndex] = useState(4);
   const qModal = useRef(null);
 
   useEffect(() => {
     if (props.results) {
       setList(props.results);
-      setMatches(props.results.slice(0, 4));
     }
-  }, [list, props.results]);
+  }, [props.results]);
+
+  useEffect(() => {
+    setMatches(list.slice(0, index));
+  }, [index, list]);
 
   const handleSubmit = (input) => {
     if (input.length >= 3) {
       const display = [];
-      matches.forEach((match) => {
+      list.forEach((match) => {
         const question = match.question_body.toLowerCase();
         if (question.includes(input.toLowerCase())) {
           display.push(match);
         }
       });
-      setMatches(display);
+      setIndex(4);
+      setList(display);
     } else {
-      setMatches(props.results.slice(0, 4));
+      setIndex(4);
+      setList(props.results);
     }
   };
 
-  // const showMoreQuestions = () => {
+  const showMoreQuestions = () => {
+    setIndex(index + 2);
+  };
+
+  const showLessQuestions = () => {
```

Comment:
Is it a requirement in the BRD for the default "see less" value to be four questions instead of decrementing by 2?